sway/input/seat.c

```diff
@@ -772,26 +772,17 @@ void seat_set_focus_warp(struct sway_seat *seat, struct sway_node *node,
 	}
 
 	if (last_focus) {
-		if (config->mouse_warping && warp &&
-				(new_output != last_output ||
-				config->mouse_warping == WARP_CONTAINER)) {
-			double x = 0;
-			double y = 0;
+		if (container && config->mouse_warping == WARP_CONTAINER && warp) {
+			cursor_warp_to_container(seat->cursor, container);
+			cursor_send_pointer_motion(seat->cursor, 0, true);
+		} else if (warp && new_output != last_output &&
+				   config->mouse_warping >= WARP_CONTAINER) {
```

Comment:
* I think you intend to use `== WARP_OUTPUT` instead of `>= WARP_CONTAINER`.
* In the warping to output part, it should just warp to the workspace and not care about containers.
* The `warp` condition exist in both statements, so it could be moved into the `if (last_focus)` one (err... why is this even conditional on having a last focus?).

Reply:
Judging by `git blame`, `last_focus` was used to remove the focus from a view previously. It will probably cause no harm to remove here. What are your thoughts on this @RyanDwyer?

Reply:
Yep, let's remove it. And by remove I mean replace it with `warp`.